packages/strapi-hook-mongoose/lib/index.js

```diff
@@ -87,17 +89,33 @@ module.exports = function(strapi) {
         connectOptions.dbName = database;
         connectOptions.useCreateIndex = true;
         connectOptions.useUnifiedTopology = useUnifiedTopology || 'false';
+        connectOptions.reconnectInterval = reconnectInterval || defaults.reconnectInterval;
+        connectOptions.reconnectTries = reconnectTries || defaults.reconnectTries;
 
-        try {
+        const connect = () => {
           /* FIXME: for now, mongoose doesn't support srv auth except the way including user/pass in URI.
            * https://github.com/Automattic/mongoose/issues/6881 */
-          await instance.connect(
-            uri ||
+          return instance.connect(
+              uri ||
               `mongodb${isSrv ? '+srv' : ''}://${username}:${encodeURIComponent(
-                password
+                  password
               )}@${host}${!isSrv ? ':' + port : ''}/`,
-            connectOptions
+              connectOptions
           );
+        }
+
+        try {
+          await connect();
+
+          instance.connection.on('disconnected', function() {
```

Comment:
Can you please use arrow functions in this code. Don't forget to implement the reconnectTries logic too :)

Reply:
I think we should leave reconnectTries logic under the mongoose hood.
As I tried to explain I cant find a reason to stay disconnected.
If you have a scenario where a strapi project could be irresponsible forever I would change my mind.

Reply:
SImple example: a docker container. if you disconnect it should exit instead of keep alive for eternity hence the retry counter

Reply:
It will not exit on disconnect. Too many listeners are bound including http port binding etc.
I use docker containers in production. No similar scenario.

Reply:
You are missing my point, whether you want a docker container to be killed or not is irrelevant.

What I am saying is that a user should be able to control that :) e.g  retry 5 times then do sth like:
```js
connection.on('someFinalEvent', ()= > { /* do sth */ }) 
```
or anything else the user wants to do.